client/components/trip/createTripModal.js

```diff
@@ -68,18 +69,24 @@ const NumberInput = (props) => {
   );
 };
 
-export const SaveTripContainer = ({ dateRange }) => {
+export const SaveTripContainer =({ dateRange }) => {
   const [isSaveModalOpen, setIsSaveModalOpen] = useState(false);
   const weatherObject = useSelector((state) => state.weather.weatherObject);
-  const search = useSelector((state) => state.search.selectedSearchResult);
```

Comment:
we need to keep the selected search bar result somewhere else i think. do we have a hook for this search already ?

Reply:
Yeah we already have but weather information was stored in state weather so that's why state.search wasn't have information regarding geoJson and we were getting errors

Reply:
previously we were accessing osm_id and other information from state.search but we were setting state.weather.selectedSearch while selecting destination

Reply:
yea we need to move that logic out of the redux store though i think ? its a bit confusing currently with both in play

Reply:
Yeah in this scenario for example we can create a small custom hook for fetching weather information instead we dispatch it first and then use it